src/release.js

```diff
@@ -22,6 +22,7 @@
  * @author Anton Tsymuk <anton@taotesting.com>
  */
 
+const fs = require('fs');
```

Comment:
I have the feeling the code isn't well organized. We are mixing the code to release a tao extention and an npm package. When we will add other types (composer libs for example) the code will become complicated and more difficult to maintain.
What about splitting in 3 files : release.js (general / adaptor), release/npm and release/tao
And when you call a method on `release` it can delegate the call to the correct implementation (adapter pattern).

Reply:
```js
module.exports = function releaseFactory(params = {}) {
  
  //determine the subject upfront
  const releaseAdaptee = require(`./release/${subject}.js`)(params);

  return {
    someGenericMethod() {
        //code is here
    },
    someSubjectDependentMethod(){
       return releaseAdaptee.someSubjectDependentMethod();
    }
  }
};
```

Reply:
The idea seems good.
I don't know if I see the need to make the Adaptee a factory taking the exact same params as its parent. I would propose to do it as a plain object containing 4 or 5 methods which are called by the releaseFactory. A few injected dependencies and return values should be all I need.

However, after sleeping on the idea I may change my mind!